src/schema/v2/FeaturedLink/createFeaturedLinkMutation.ts

```diff
@@ -0,0 +1,100 @@
+import {
+  GraphQLString,
+  GraphQLUnionType,
+  GraphQLObjectType,
+  GraphQLBoolean,
+} from "graphql"
+import { mutationWithClientMutationId } from "graphql-relay"
+import { GraphQLNonNull } from "graphql"
+import { ResolverContext } from "types/graphql"
+import { FeaturedLinkType } from "./featuredLink"
+import {
+  formatGravityError,
+  GravityMutationErrorType,
+} from "lib/gravityErrorHandler"
+
+interface GravityInput {
```

Comment:
Do we need to also include the ability to pass a `gemini_token` arg? If this is explicitly passed, or gets associated elsewhere in gravity is not clear to me at this point...

Reply:
I chose not to add that b/c that field is generated only during the image-processing/upload flow. So at some point, yes, that field definitely needs to be set on the object (via a mutation).

So, we could add that here and use/re-use this mutation to set that field during the image-processing flow, but perhaps our image processing flow might use a mutation that sets it there? Not sure!

This is one of those threads/attributes that could be followed up on once the front-end use case is a bit clearer I think.

Reply:
Another idea is a Gravity update which would allow this association to happen in Gravity (vs. more orchestrated on the client) could work. I think artworks and their images (who have a similar token field) work in that way now.

Reply:
I believe I need this for the [adding a feature](https://www.figma.com/file/RiH2Y6TKOPG1kyEWdkJ0Zw/Forque-Platform-redesign?node-id=169%3A26297&t=ylNMKfXTr39X51dW-4) link flow that i'm currently working on, in order to associate the user uploaded `image` with the feature link being created unless I'm just misunderstanding...?

Reply:
Right, but I wasn't sure how you were implementing that. One option (which is I think what you're doing/saying) could be that the FE component which is orchestrating things knows what the token is, by virtue of making an upload/process call, and sends that data to Gravity (in which case likely adding it to these existing mutations is warranted).

But, a _possible_ approach (which if you already have the first one working its moot!), would be to send the uploaded file info to Gravity in the API call (`source_bucket` and `source_key`), and let Gravity make the Gemini API call and persist the token! I believe this is how artwork/image uploading works - the FE sends back the location of the upload, and Gravity does the rest - https://github.com/artsy/gravity/blob/16d524b81d1a2ca657e77db9e132cf7e277947f0/app/models/artwork_image_builder.rb#L15